test/core/api/client/ws_client.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "core/api/client/ws_client.hpp"
+
+#include <boost/beast/http/status.hpp>
+
+OUTCOME_CPP_DEFINE_CATEGORY(test, WsClientError, e) {
+  using test::WsClientError;
+  switch (e) {
+    case WsClientError::CONNECTION_FAILED:
+      return "connection failed";
+    case WsClientError::WEBSOCKET_ERROR:
```

Comment:
Er, these seem not much informative to me, is there a way to pass lower-level more informative errors outside?

Reply:
We can, but on this stage error means websocket handshake error. It's anough for us.